fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/shim/ChaincodeBase.java

```diff
@@ -442,53 +413,48 @@ protected final void processCommandLineOptions(final String[] args) {
      * set fields from env.
      */
     public final void processEnvironmentOptions() {
+        this.id = System.getenv().getOrDefault(CORE_CHAINCODE_ID_NAME, this.id);
+        
+        Optional.ofNullable(System.getenv(CORE_PEER_ADDRESS))
+                .ifPresent(address -> {
+                    String[] hostArr = address.split(":");
+                    if (hostArr.length == 2) {
+                        this.port = Integer.parseInt(hostArr[1].trim());
+                        this.host = hostArr[0].trim();
+                    } else {
+                        LOGGER.severe(() -> String.format("peer address argument should be in host:port format, ignoring current %s", address));
+                    }
+                });
```

Comment:
I don't see any real benefit to using an optional only to perform a null check

Reply:
Replaced the Optional.ofNullable().ifPresent() with a simple null check using an if statement. Improved readability by using a more straightforward and familiar Java construct.